app/models/katello/repository.rb

```diff
@@ -103,6 +103,8 @@ class Repository < Katello::Model
     scope :non_archived, where('environment_id is not NULL')
     scope :archived, where('environment_id is NULL')
 
+    scope :completer_scope, lambda { |options| self.send(options[self.table_name]) }
```

Comment:
I think i'd rather not depend on named scopes and support raw lambdas.  Sorry i had mentioned it to ehelms on irc, forgot to type it here.